tekton/ci/shared/go_coverage_task.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: go-coverage
+  annotations:
+    description: |
+      Clones a repository from github into an empty-dir workspace.
+spec:
+  params:
+    - name: prNumber
+      description: The number of the pull request to check out.
+    - name: branchName
+      description: Branch of the pull request to check out.
+    - name: pullSha
+      description: sha of the commit making the pull request
+    - name: baseSha
+      description: sha of the commit making the pull request against.
+    - name: jobType
+      description: Type of prow job - presubmit/postsubmit/periodic.
+    - name: buildNumber
+      description: Prow type build ID.
+    - name: repoName
+      description: name of the repository
+    - name: repoOwner
+      description: owner of the repo (user/orgaization).
+    - name: jobName
+      description: name of the job.
+  steps:
+    - name: clone-repo
+      image: alpine/git
+      script: |
+        #!/bin/sh
+        git clone https://github.com/$(params.repoOwner)/$(params.repoName).git
+        cd $(params.repoName)
+        git fetch origin pull/$(params.prNumber)/head:$(params.branchName)
+        git checkout $(params.branchName)
+      workingDir: "$(workspaces.shared-workspace.path)"
+    - name: run-coverage
+      image: gcr.io/tekton-releases/dogfooding/coverage:latest
+      env:
+        - name: PULL_PULL_SHA
+          value: $(params.pullSha)
+        - name: PULL_BASE_SHA
+          value: $(params.baseSha)
+        - name: PULL_NUMBER
+          value: $(params.prNumber)
+        - name: JOB_TYPE
+          value: $(params.jobType)
+        - name: BUILD_NUMBER
+          value: $(params.buildNumber)
+        - name: REPO_NAME
+          value: $(params.repoName)
+        - name: REPO_OWNER
+          value: $(params.repoOwner)
+        - name: JOB_NAME
+          value: $(params.jobName)
+      command:
+        - /coverage
+      args:
+        - "--postsubmit-gcs-bucket=tekton-prow"
```

Comment:
How does the auth for the gcs bucket work? Does the SA for the run have to have permissions to post to this bucket?

Reply:
I think the gcs bucket is public so anyone has read permissions. eg. I was always able to do something like `gustil ls gs://tekton-prow`

Reply:
The coverage tool fetches the `post-submit` run. It does not write to it. But let me double check this.

Reply:
Interesting. When I open a docker container on my machine and try to run the coverage command, I get this error:
```
2022/10/20 19:37:46 Running workflow: presubmit
2022/10/20 19:37:46 Failed to create client: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
```

But, if I simply launch the task on my cluster on GKE, I don't see that. Is it possible that because I'm running the cluster on GKE, my credentials are some how available?

Reply:
yeah, it's using ADC, so it seems tied to Google: https://cloud.google.com/docs/authentication#adc